ui/components/app/assets/token-cell/token-cell-list-item.tsx

```diff
@@ -0,0 +1,399 @@
+import React, { useContext, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import classnames from 'classnames';
+import { getNativeTokenAddress } from '@metamask/assets-controllers';
+import { Hex } from '@metamask/utils';
+import {
+  BackgroundColor,
+  BlockSize,
+  Display,
+  FlexDirection,
+  FontWeight,
+  IconColor,
+  JustifyContent,
+  TextAlign,
+  TextColor,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import {
+  AvatarNetwork,
+  AvatarNetworkSize,
+  AvatarToken,
+  BadgeWrapper,
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  ButtonSecondary,
+  IconName,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  SensitiveText,
+  SensitiveTextLength,
+  Text,
+} from '../../../component-library';
+import {
+  getMarketData,
+  getCurrencyRates,
+  getNativeCurrencyForChain,
+} from '../../../../selectors';
+import {
+  getImageForChainId,
+  getMultichainIsEvm,
+} from '../../../../selectors/multichain';
+import Tooltip from '../../../ui/tooltip';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { MetaMetricsContext } from '../../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../../shared/constants/metametrics';
+import {
+  CURRENCY_SYMBOLS,
+  NON_EVM_CURRENCY_SYMBOLS,
+} from '../../../../../shared/constants/network';
+import { hexToDecimal } from '../../../../../shared/modules/conversion.utils';
+
+import { NETWORKS_ROUTE } from '../../../../helpers/constants/routes';
+import { setEditedNetwork } from '../../../../store/actions';
+import {
+  SafeChain,
+  useSafeChains,
+} from '../../../../pages/settings/networks-tab/networks-form/use-safe-chains';
+import { NETWORK_TO_SHORT_NETWORK_NAME_MAP } from '../../../../../shared/constants/bridge';
+import { getNetworkConfigurationsByChainId } from '../../../../../shared/modules/selectors/networks';
+import { TokenFiatDisplayInfo } from '../types';
+import { PercentageChange } from '../../../multichain/token-list-item/price/percentage-change';
+import { StakeableLink } from '../../../multichain/token-list-item/stakeable-link';
+
+type TokenListItemProps = {
+  className?: string;
+  token: TokenFiatDisplayInfo;
+  onClick?: (arg?: string) => void;
+  showPercentage?: boolean;
+  // Figure out how to abstract the bottom from Asset.tsx
+  tooltipText?: string;
+  isTitleNetworkName?: boolean;
+  isTitleHidden?: boolean;
+  isPrimaryTokenSymbolHidden?: boolean;
```

Comment:
These are props from the previous implementation. They were accepting props from `asset-picker-amount/Asset.tsx`

The plan is to remove these props to make rendering the token list easier